JANUSGRAPH-1866 Filter out only system vertices in Hadoop Vertex dese…

YevIgn opened the pull request:
Fixes #1866 

Signed-off-by: Evgeniy Ignatiev <yevgeniy.ignatyev@gmail.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[doc only]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

FlorianHockmann commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity decreasing per file
==============================
+ janusgraph-hadoop/src/main/java/org/janusgraph/hadoop/formats/util/JanusGraphVertexDeserializer.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=4658350&bid=15599418)

FlorianHockmann commented on janusgraph-hadoop/src/main/java/org/janusgraph/hadoop/formats/util/JanusGraphVertexDeserializer.java:
```diff
@@ -25,6 +25,7 @@
 import org.janusgraph.graphdb.internal.InternalRelationType;
 import org.janusgraph.graphdb.relations.RelationCache;
 import org.janusgraph.graphdb.types.TypeInspector;
+import org.janusgraph.graphdb.types.system.BaseKey;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused imports such as 'org.janusgraph.graphdb.types.system.BaseKey'](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=4658350)

YevIgn replied:
Removed.

FlorianHockmann commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ janusgraph-hadoop/src/main/java/org/janusgraph/hadoop/formats/util/JanusGraphVertexDeserializer.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=4658350&bid=15599418)

FlorianHockmann commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ janusgraph-hadoop/src/main/java/org/janusgraph/hadoop/formats/util/JanusGraphVertexDeserializer.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=4658350&bid=15599418)

YevIgn commented:
Added test for skipping schema vertices. Added correct HBase Snapshot tests.

But I see that there are some failures that are completely unrelated to the PR - https://travis-ci.org/JanusGraph/janusgraph/jobs/624525059?utm_medium=notification&utm_source=github_status

org.janusgraph.diskstorage.cassandra.embedded.EmbeddedLogTest.testLogIsDurableAcrossReopenSerial has no relationship with the code I have changed, some methods of this test are already marked as flay, maybe this one is flaky too?

porunov commented:
Thank you @YevIgn . I have added an issue about the fluky test #1908 . The tests are now restarted

farodin91 commented:
@YevIgn Could you squash and rebase your commits?

YevIgn commented:
@farodin91 Done.

FlorianHockmann commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ janusgraph-hadoop/src/main/java/org/janusgraph/hadoop/formats/util/JanusGraphVertexDeserializer.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/JanusGraph/janusgraph/pullRequest?prid=4658350&bid=15599418)
